gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -568,8 +576,10 @@ MacroBuiltin::include_str_handler (Location invoc_locus,
   if (expect_single)
     rust_error_at (invoc_locus, "%s was not a valid utf-8 file",
 		   target_filename.c_str ());
-  else
+  else if (!bytes.empty ())
     str = std::string ((const char *) &bytes[0], bytes.size ());
+  else
+    str = "";
```

Comment:
@CohenArthur is that the correct way to "do nothing"? Is there a better way to create an empty Fragment?

Reply:
I think this is the best way. I'll have a think about it but it seems good to me

Reply:
I've experimented with the optional approach and have refactored a bit. Tests are all OK.

Reply:
Note that the initial question of "how to create empty fragment" had still to be answered. I've created a `create_empty` next to the error one.

Reply:
You could try replacing ```(const char *) &bytes[0]``` with ```(const char *) bytes.data()```